amazon/ion/simpleion.py

```diff
@@ -372,7 +372,35 @@ def add(obj):
         event = reader.send(NEXT_EVENT)
 
 
-def loads(fp, encoding='utf-8', cls=None, object_hook=None, parse_float=None, parse_int=None, parse_constant=None,
-          object_pairs_hook=None, use_decimal=None, **kw):
-    """Not yet implemented"""
-    raise IonException("Not yet implemented")
+def loads(ion_str, catalog=None, single_value=True, encoding='utf-8', cls=None, object_hook=None, parse_float=None,
+          parse_int=None, parse_constant=None, object_pairs_hook=None, use_decimal=None, **kw):
+    """Deserialize ``ion_str``, which is a string representation of an Ion object, to a Python object using the
+    conversion table used by load (above).
+
+    Args:
+        fp (str): A string representation of Ion data.
+        catalog (Optional[SymbolTableCatalog]): The catalog to use for resolving symbol table imports.
+        single_value (Optional[True|False]): When True, the data in ``ion_str`` is interpreted as a single Ion value,
+            and will be returned without an enclosing container. If True and there are multiple top-level values in
+            the Ion stream, IonException will be raised. NOTE: this means that when data is dumped using
+            ``sequence_as_stream=True``, it must be loaded using ``single_value=False``. Default: True.
+        encoding: NOT IMPLEMENTED
+        cls: NOT IMPLEMENTED
+        object_hook: NOT IMPLEMENTED
+        parse_float: NOT IMPLEMENTED
+        parse_int: NOT IMPLEMENTED
+        parse_constant: NOT IMPLEMENTED
+        object_pairs_hook: NOT IMPLEMENTED
+        use_decimal: NOT IMPLEMENTED
+        **kw: NOT IMPLEMENTED
+
+    Returns (Any):
+        if single_value is True:
+            A Python object representing a single Ion value.
+        else:
+            A sequence of Python objects representing a stream of Ion values.
+    """
+    ion_buffer = BytesIO(ion_str.encode())
```

Comment:
Does this work for both binary and text types? This API should be able to work for both; users should be able to use this to read both text and binary Ion data, e.g.

`res = loads(b'\xE0\x01\x00\xEA\x20')`

and

`res = loads(u'$ion_1_0 0')`

Symmetrically, perhaps `dumps` should work for both too. The tricky part is keeping consistent with `simplejson.dumps`, which looks like its return type changed from a binary type to a text type from Python 2 -> 3 (because it remained `str` the whole time). Maybe we can just return a binary type for both text and binary Ion? Or return a text type for text Ion? Thoughts?

Reply:
Huh, I didn't know simplejson had implicit behavior like that. Seems kinda weird, but then I've never used the library myself. Tracing through the code, it has [this decoder class](https://github.com/simplejson/simplejson/blob/master/simplejson/decoder.py#L363-L400) thing going on. I can't tell where the hex conversion happens though.

Is a byte input expected to be hex encoded like this? How does the `encoding` parameter play into this, if at all?